staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -194,11 +195,28 @@ func (d *Helper) DeleteOrEvictPods(pods []corev1.Pod) error {
 
 func (d *Helper) evictPods(pods []corev1.Pod, policyGroupVersion string, getPodFn func(namespace, name string) (*corev1.Pod, error)) error {
 	returnCh := make(chan error, 1)
-
+	// 0 timeout means infinite, we use MaxInt64 to represent it.
+	var globalTimeout time.Duration
+	if d.Timeout == 0 {
+		globalTimeout = time.Duration(math.MaxInt64)
+	} else {
+		globalTimeout = d.Timeout
+	}
+	ctx, cancel := context.WithTimeout(context.TODO(), globalTimeout)
```

Comment:
Not related to the management of the goroutines, but should the value of globalTimeout also have an impact on the timeout passed to `waitForDelete` as well? It feels a bit odd that we could extend the timeout based on sitting in the call to `waitForDelete`

Reply:
That's what I thought too.  But actually, the bit at the bottom also catches the timeout and waitForDelete is closed for us.

Reply:
@detiber Looks like there was a goroutine leak here as well.  I updated.